pkg/controller/build/build_controller.go

```diff
@@ -84,8 +85,10 @@ func (r *ReconcileBuild) Reconcile(request reconcile.Request) (reconcile.Result,
 
 	b := &build.Build{}
 	err := r.client.Get(context.Background(), request.NamespacedName, b)
-	if err != nil {
+	if err != nil && !apierrors.IsNotFound(err) {
```

Comment:
I´m not sure I understand the need on using the `apierrors` . I saw this before on the code, can you explain. I will only return the error as in:
```
if err := r.client.Get(context.Background(), request.NamespacedName, b); err != nil{
		return reconcile.Result{}, err
}
```

Reply:
I learned this kind of refined logic from other operators, I think it is also better:

If we want to reconcile a resource, but this resource doesn't exist, I think we don't need to reconcile it anymore.

It may because some bad resources or some hack I think.

You can see from many places:
- [A google book] https://books.google.com/books?id=I_3RDwAAQBAJ&pg=PT77&lpg=PT77&dq=reconcile+errors.IsNotFound&source=bl&ots=3klX1AoKR_&sig=ACfU3U1PEu6U3v5qJ8eTIcTrw-Ye6mZ6CQ&hl=en&sa=X&ved=2ahUKEwjAjraf3oPpAhVymK0KHU0WD9MQ6AEwAHoECAkQAQ#v=onepage&q=reconcile%20errors.IsNotFound&f=false
- [operator-sdk examples] https://github.com/operator-framework/operator-sdk/blob/master/doc/user-guide.md#add-a-new-controller
- [kpack code] https://github.com/pivotal/kpack/blob/master/pkg/reconciler/v1alpha1/build/build.go#L78-L81

Reply:
alright, nice to know, thanks